<a href="https://colab.research.google.com/github/alerestrepo185/alerestrepo185.github.io/blob/main/PROYECTO_2_(MACHINE_LEARNING)_RESTREPO%2C_SANTAMARIA%2C_YEPES.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **PROYECTO #2**

Paula Alejandra Restrepo Velásquez

Andrés Felipe Santamaria Robayo

Junior Yepes Vasquez

In [ ]:
!pip install wooldridge
import wooldridge as wd

In [ ]:
df=wd.data ("wage1")
wage = df["wage"]

In [ ]:
wage

0       3.10
1       3.24
2       3.00
3       6.00
4       5.30
       ...  
521    15.00
522     2.27
523     4.67
524    11.56
525     3.50
Name: wage, Length: 526, dtype: float64

In [ ]:
class Proyecto2:
  def __init__(self, wage):
    self.wage = wage
  def media (self):
    return sum(self.wage)/len(self.wage)
  def mediana (self):
    data = sorted (self.wage)
    ale = len (data) // 2
    return (data [ale - 1] + data [ale]) / 2 #datos pares
  def desviacion (self):
    vector=[]
    media=self.media()
    for i in (self.wage):
      vector.append((i-media)**2)
    return ((sum(vector)/(len(self.wage)-1))**(1/2))
  def varianza (self):
    desviacion=self.desviacion()
    return desviacion**(2)
  def coeficiente (self):
    media=self.media()
    desviacion=self.desviacion()
    return desviacion/media

In [ ]:
Equipo_1 = Proyecto2 (wage)

In [ ]:
Equipo_1.media()

5.896102674787035

- El valor arrojado 5.896102674787035 refiere a que el promedio de los salarios por hora  registrados en esta base de datos es de  5.89 dolares.


In [ ]:
Equipo_1.mediana()

4.650000095367432

- Ordenando los datos,  el salarío por hora que se encuentra el punto medio de los datos es de 4.65 dolares por hora.

In [ ]:
Equipo_1.desviacion()

3.693086021713414

- La desviación de 3.69 refiere a que existe una dispersión media de 3.69 dolares por hora respecto al promedio del salario que es 5.89 dolares.

In [ ]:
Equipo_1.varianza()

13.63888436377501

- La varianza registrada de 13.63 representa una una alta  variabilidad de los datos respecto al valor promedio; es decir, existe una gran variabilidad referente al salario para la población encuestada.

In [ ]:
Equipo_1.coeficiente()

0.6263605343078268

- Este  coeficinete de variación  x 100% nos indica que existe una variación de 62.63% de los datos respecto a su media; es decir, respecto a que el salario medio por hora es 5.89, los salarios de esta muestra varian un 62.63%,encontrando así una gran heterogeneidad en el nivel salarial para la población encuestada.

In [ ]:
df=wd.data ("wage1")
educ = df["educ"]

In [ ]:
educ

0      11
1      12
2      11
3       8
4      12
       ..
521    16
522    10
523    15
524    16
525    14
Name: educ, Length: 526, dtype: int64

In [ ]:
class Proyecto22:
  def __init__(self, educ):
    self.educ = educ
  def media (self):
    return sum(self.educ)/len(self.educ)
  def mediana (self):
    data = sorted (self.educ)
    ale = len (data) // 2
    return (data [ale - 1] + data [ale]) / 2 #datos pares
  def desviacion (self):
    vector=[]
    media=self.media()
    for i in (self.educ):
      vector.append((i-media)**2)
    return ((sum(vector)/(len(self.educ)-1))**(1/2))
  def varianza (self):
    desviacion=self.desviacion()
    return desviacion**(2)
  def coeficiente (self):
    media=self.media()
    desviacion=self.desviacion()
    return desviacion/media

In [ ]:
Equipo_11 = Proyecto22 (educ)

In [ ]:
Equipo_11.media()

12.562737642585551

- El promedio de años de escolaridad para esta muestra es de 12.56 años.

In [ ]:
Equipo_11.mediana()

12.0

- Ordenando los datos, se encuentra que los años de escolaridad que se encuentran en el punto medio de la totalidad de los datos  es de 12.0. 

In [ ]:
Equipo_11.desviacion()

2.7690224019436975

- Este valor refiere a que en la muestra existe en el nivel de escolarización  una variación media de 2.76 años respecto al valor promedio de 12.0 años de escolaridad.

In [ ]:
Equipo_11.varianza()

7.667485062466044

- Este valor indica que existe una alta variabilidad en los años escolares registrados en esta población estudiada.

In [ ]:
Equipo_11.coeficiente()

0.22041552412566356

- Este valor multiplicado x 100% indica que existe una variación de 22.04% de los datos respceto a su media; es decir, si el promedio de años de escolaridad es de 12.0 años , en la variable los años varian en  22.04 %, resaltando así una variación significativa en esta población estudiada.